### PR TITLE
Dictionaries ddl parser

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Other

Short description (up to few sentences):
Parser for dictionaries DDL without any semantic.

### CODE REVIEW

File: dbms/src/Parsers/ASTDictionary.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
+#include <Parsers/ASTDictionary.h>
```
Comment (alexey-milovidov) [inline dbms/src/Parsers/ASTDictionary.cpp]: I thought that we can parse free-form syntax, convert to DOM and re-use existing code.

Comment (alexey-milovidov) [reply dbms/src/Parsers/ASTDictionary.cpp]: It can be better, because <source> can have custom elements depend on the source itself.

Comment (alesapin) [reply dbms/src/Parsers/ASTDictionary.cpp]: Yes, I also plan to convert AST to DOM (Poco::AbstractConfiguration) and reuse the current code in the ExternalLoader. `SOURCE` is the only section where we have absolutely custom properties and it's already represented as [function with key-value arguments](https://github.com/ClickHouse/ClickHouse/pull/7209/files/0e664807e9029b94c3a25b2b9bd0a61626a3de5b#diff-65bc97a2f72ddc9a11333338f2a7b4efR59) which can hold any amount of custom properties (also nested). 

Or you mean some general conversion function?

**Comment by alesapin** [comment]
Docs check failed on master

**Comment by BayoNet** [comment]
Do we need any docs here?

**Pull request merged**